fvm/errors/errors.go

```diff
@@ -4,22 +4,47 @@ import (
 	stdErrors "errors"
 	"fmt"
 
-	"github.com/hashicorp/go-multierror"
 	"github.com/onflow/cadence/runtime"
 	"github.com/onflow/cadence/runtime/errors"
 )
 
 type Unwrappable interface {
+	error
 	Unwrap() error
 }
 
+type UnwrappableErrors interface {
+	error
+	Unwrap() []error
+}
+
 type CodedError interface {
 	Code() ErrorCode
 
 	Unwrappable
 	error
 }
 
+// CodedFailure is a subtype of CodedError specifically for failures
+// This is used to make type checking with `errors.As` and `errors.Is` easier and more accurate
+type CodedFailure interface {
+	Code() FailureCode
+
+	Unwrappable
+	error
+}
+
+// WrappedParentError is a wrapper for errors.ParentError to make it conform to the unwrappable
+// interface used by the std errors lib. Once these error types are updated, this can be removed.
+// See https://github.com/onflow/cadence/issues/3035
```

Comment:
Can we maybe just fix this directly in Cadence, so we don't need special handing here in FVM?

Reply:
Yes, we should fix this in cadence. I just added it so I could move forward while that issue is addressed. the extra logic is pretty minimal and can be removed easily once that's ready.

Reply:
https://github.com/onflow/cadence/pull/3052